### DISCUSSION TITLE
Valid path of XML element 'inspire_dls:ExtendedCapabilities' in WCS service?

### DISCUSSION THREAD

**Discussion by milanSch** [INSPIRE-MIF/helpdesk-validator#920] [Q&A] [answered]
Dear, 

1. I created WCS service 2.0.1 using Geoserver 2.20.0 + Geoserver INSPIRE extension.
2. Geoserver INSPIRE extension created 'inspire_dls:ExtendedCapabilities' element in path 
`/wcs:Capabilities/wcs:Contents/wcs:Extension/ows:ExtendedCapabilities/inspire_dls:ExtendedCapabilities'`

![2021-11-15_090443](https://user-images.githubusercontent.com/4444428/141748482-34762a44-3114-4392-86c0-cbef44d6c3c2.png)
3. I tested WCS service using https://inspire.ec.europa.eu/validator/home/index.html - Download Service - Web Coverage Service (WCS)
and there is reported error:
`'The ExtendedCapabilities section shall be provided in the path: /Capabilities/OperationsMetadata/ExtendedCapabilities/ExtendedCapabilities'`
4. I read document 'Technical Guidance for the implementation of INSPIRE Download Services using Web Coverage Services (WCS) ' https://inspire.ec.europa.eu/id/document/tg/download-wcs
On page 33: 'In order to make the Download Service INSPIRE metadata elements available via a standard WCS it is necessary to use ows:ExtendedCapabilites in the WCS capabilities response and publish the INSPIRE metadata according to an extension schema within an  inspire_dls:ExtendedCapabilities element.'
5.  I asked here https://gis.stackexchange.com/questions/415766/modify-replace-getcapabilities-response-geoserver-inspire-extension-wcs
6. I created two files with GetCapabilities response. Just to test if they are valid. Both documents are valid using validation tool implemented in XMLspy.
-  first - 'inspire_dls:ExtendedCapabilities' is in this path
/wcs:Capabilities/wcs:Contents/wcs:Extension/ows:ExtendedCapabilities/inspire_dls:ExtendedCapabilities 
-  second - 'inspire_dls:ExtendedCapabilities' is in this path
/wcs:Capabilities/ows:OperationsMetadata/ows:ExtendedCapabilities/inspire_dls:ExtendedCapabilities

QUESTIONS: Please, do you know in which position/path should be WCS service 'ows:ExtendedCapabilites' element placed? Are both paths possible?

Thank you
milan

**Comment by dperezBM** [comment]
Dear @milanSch,

Thank you for your message. We will analyze this issue and we will get back to you with some feedback.

Best regards

**Comment by klimeto** [comment]
Dears,

Reading this documentation: https://github.com/inspire-eu-validation/download-wcs/blob/1.0/wcs-core/at08-list-supported-languages.md

I would guess the XML fragment of extended capabilities should be encoded as following: 

```xml
<wcs:Capabilities>
	<ows:OperationsMetadata>
		<ows:ExtendedCapabilities>
			<inspire_dls:ExtendedCapabilities>
				<inspire_common:MetadataUrl>
					<inspire_common:URL>https://rpi.gov.sk/rpi_csw/service.svc/get?request=GetRecordById&service=CSW&version=2.0.2&elementSetName=full&outputschema=http://www.isotc211.org/2005/gmd&Id=https://data.gov.sk/set/rpi/gmd/00626031/CF3DFCB3AEFC23F1E0530210000A4780IMA1990</inspire_common:URL>
					<inspire_common:MediaType>application/vnd.ogc.csw.GetRecordByIdResponse_xml</inspire_common:MediaType>
				</inspire_common:MetadataUrl>
				<inspire_common:SupportedLanguages>
					<inspire_common:DefaultLanguage>
						<inspire_common:Language>slo</inspire_common:Language>
					</inspire_common:DefaultLanguage>
				</inspire_common:SupportedLanguages>
				<inspire_common:ResponseLanguage>
					<inspire_common:Language>slo</inspire_common:Language>
				</inspire_common:ResponseLanguage>
				<inspire_dls:SpatialDataSetIdentifier metadataURL="https://rpi.gov.sk/rpi_csw/service.svc/get?request=GetRecordById&service=CSW&version=2.0.2&elementSetName=full&outputschema=http://www.isotc211.org/2005/gmd&Id=https://data.gov.sk/set/rpi/gmd/00626031/4E34D8FA16B66BFFE0530110000A3369IMA1990">
					<inspire_common:Code>https://data.gov.sk/set/rpi/dat/00626031/4E34D8FA16B66BFFE0530110000A3369IMA1990</inspire_common:Code>
					<inspire_common:Namespace>https://data.gov.sk/set/rpi/dat/00626031/</inspire_common:Namespace>
				</inspire_dls:SpatialDataSetIdentifier>
			</inspire_dls:ExtendedCapabilities>
		</ows:ExtendedCapabilities>
	</ows:OperationsMetadata>
</wcs:Capabilities>
```

@dperezBM can you confirm or update?

Thanks

**Comment by KathiSchleidt** [comment]
I agree with Klimeto - the following path utilized in the rasdaman implementation seems to fit what the validator expects:
/wcs:Capabilities/ows:OperationsMetadata/ows:ExtendedCapabilities/inspire_dls:ExtendedCapabilities

Thus assuming that your issue is a glitch in GeoServer.

The only validation issue I'm currently getting on the following endpoint pertains to the AcceptLanguages parameter in the GetCapabilities request - based on the WCS TG, this is a recommendation, but the validator marks it as an error, details at #683 :

https://coverage-demo.wetransform.eu/rasdaman/ows#/services

**Comment by arantzaetxebarria** [comment, marked as answer]
Dear @milanSch 

As @klimeto and @KathiSchleidt have pointed out, for now, the correct position of the ExtendedCapabilities element is 
/wcs:Capabilities/ows:OperationsMetadata/ows:ExtendedCapabilities/inspire_dls:ExtendedCapabilities.
There is a discussion https://github.com/INSPIRE-MIF/helpdesk-validator/issues/56, which is still open.

Regards

**Comment by KathiSchleidt** [comment]
To my view, this thread (as well as #56) can both be closed as resolved. If data providers wish to utilize the `wcs:Contents/wcs:Extension` option, a new issue with a change request (ideally with exemplary implementation) should be provided.

**Comment by KathiSchleidt** [comment]
A short note that we've now solved all problems with validating rasdaman WCS, details at: https://www.geoconnexion.com/news/rasdaman-validated-as-inspire-compliant-wcs
